test/cmd/apply.sh

```diff
@@ -268,6 +268,29 @@ __EOF__
   # cleanup
   kubectl delete ns nsb
 
+  ## kubectl apply --prune --all --namespace NS can only prune resources in the namespace
+  # Pre-Condition: namespace nsb exists; no POD exists
+  kubectl create ns nsb
+  kube::test::get_object_assert pods "{{range.items}}{{${id_field:?}}}:{{end}}" ''
+  # apply a into namespace nsb
+  kubectl apply --namespace nsb -f hack/testdata/prune/a.yaml "${kube_flags[@]:?}"
+  kube::test::get_object_assert 'pods a -n nsb' "{{${id_field:?}}}" 'a'
+  # apply b with namespace
+  kubectl apply --namespace nsb -f hack/testdata/prune/b.yaml "${kube_flags[@]:?}"
+  kube::test::get_object_assert 'pods b -n nsb' "{{${id_field:?}}}" 'b'
+  # apply non-reapable type
+  kubectl apply --all --prune -f hack/testdata/prune-reap/a.yml 2>&1 "${kube_flags[@]:?}"
+  kube::test::get_object_assert 'pvc a-pvc' "{{${id_field:?}}}" 'a-pvc'
+  # apply --prune must prune a
+  kubectl apply --prune --all --namespace nsb -f hack/testdata/prune/b.yaml
+  # check wrong pod doesn't exist and right pod exists
+  kube::test::wait_object_assert 'pods -n nsb' "{{range.items}}{{${id_field:?}}}:{{end}}" 'b:'
```

Comment:
What happened to b.yaml resources?

Reply:
resources in b.yml don't get removed. The line asserts that there is only 'b' pod in the namespace 'nsb'